pkg/datapath/linux/ipsec/ipsec_linux.go

```diff
@@ -271,27 +271,29 @@ func UpsertIPsecEndpoint(local, remote *net.IPNet, dir IPSecDir) (uint8, error)
 	return spi, nil
 }
 
-func decodeIPSecKey(keyRaw string) ([]byte, error) {
+func decodeIPSecKey(keyRaw string) (int, []byte, error) {
 	// As we have released the v1.4.0 docs telling the users to write the
 	// k8s secret with the prefix "0x" we have to remove it if it is present,
 	// so we can decode the secret.
 	keyTrimmed := strings.TrimPrefix(keyRaw, "0x")
-	return hex.DecodeString(keyTrimmed)
+	key, err := hex.DecodeString(keyTrimmed)
+	return len(keyTrimmed), key, err
 }
 
 // LoadIPSecKeysFile imports IPSec auth and crypt keys from a file. The format
 // is to put a key per line as follows, (auth-algo auth-key enc-algo enc-key)
-func LoadIPSecKeysFile(path string) (uint8, error) {
+func LoadIPSecKeysFile(path string) (int, uint8, error) {
```

Comment:
nit: Might be useful to document this new return value (what it is, units).

Reply:
Added a comment.